libstuff/SHTTPSManager.cpp

```diff
@@ -239,3 +240,57 @@ bool SStandaloneHTTPSManager::_onRecv(Transaction* transaction)
     transaction->response = getHTTPResponseCode(transaction->fullResponse.methodLine);
     return false;
 }
+
+list<SStandaloneHTTPSManager::Transaction*> SStandaloneHTTPSManager::_httpsSendMultiple(const string& url, vector<SData>& sendRequests)
+{
+    // Synchronize dequeuing requests, and saving results.
+    int currentIndex = 0;
+
+    string host, path;
+    if (!SParseURI(url, host, path)) {
+        _activeTransactionList.push_front(_createErrorTransaction());
+        return _activeTransactionList;
```

Comment:
This is wrong. What if there are existing transactions?

Also, I'm not sure `push_front` works in general in this function.

If we `push_front` all the new transactions, then I think we end up reading all the results in backwards order here:
https://github.com/Expensify/Bedrock/blob/6393f52183d5280507b583d2f9dc044d403ffc3c/libstuff/SHTTPSManager.cpp#L116

Have you tested that the responses you get are the correct ones in the correct order?

Reply:
I chose to use `push_front` to keep it consistent with what we do in `httpsSend`.
https://github.com/Expensify/Bedrock/blob/6393f52183d5280507b583d2f9dc044d403ffc3c/libstuff/SHTTPSManager.cpp#L231-L234

I just tested, it, however, and you are correct that it does in fact mess up the order of the results.

I'm not sure how/why this works correctly for `httpsSend`. If I'm going to refactor `_httpsSendMultiple` to call `_httpsSend` in a loop, then I'm going to have to change the `push_front` in `_httpsSend` to a `push_back`...and that makes me nervous.

Reply:
Disregard ^that comment. It does not come in the correct order. I found out why, and why we have been able to use `_httpsSend` with `push_front` up until now.

Since each socket only made one request, deserialization of the buffer only needed to be matched to whatever transaction that socket was made for, here:
https://github.com/Expensify/Bedrock/blob/6393f52183d5280507b583d2f9dc044d403ffc3c/libstuff/SHTTPSManager.cpp#L129-L134

I'm fairly certain that we should be able to change the `push_front`s to `push_back`s without messing up any place that currently calls `_httpsSend`.